src/views/activity/daily/ActivityDaily.vue

```diff
@@ -209,7 +224,7 @@ export default {
     },
     setDate: function(date, periodLength) {
       const new_date = moment(date)
-        .startOf(periodLength)
+        .startOf(this.periodLengthMoment)
```

Comment:
Don't use `this.periodLengthMoment` as it relies on `this.periodLength`, which makes it not respect the function parameter.